### ISSUE TITLE
Require RegisterOrg tx author to be a registered user

### ISSUE THREAD

**Issue by NunoAlexandre** [radicle-dev/radicle-registry#397]
### Context

Found that we miss respecting the validation _"α is associated with a registered user u."_ when registering an org while discussing solutions for https://github.com/radicle-dev/radicle-registry/issues/396.

![Screenshot from 2020-04-24 09-51-30](https://user-images.githubusercontent.com/7075260/80188348-53fee080-8611-11ea-95e2-89a233891faf.png)

_Originally posted by @NunoAlexandre in https://github.com/radicle-dev/radicle-registry/issues/396#issuecomment-618862275_

### Changes

When registering an org:

1. we require there to be a registered user with an account id equal to the public key of the tx author and fail otherwise.
2. store the user id as the single org member. This requires changing Org::members to be a list of Ids instead of a list of account ids.

**Comment by geigerzaehler** [comment]
@NunoAlexandre: Could you please clarify what part of the `register-org` specification needs to be fixed and how that would look like? This is especially important since the spec itself is not clear on the behavior (https://github.com/radicle-dev/registry-spec/issues/18).

**Comment by NunoAlexandre** [comment]
> @NunoAlexandre: Could you please clarify what part of the `register-org` specification needs to be fixed and how that would look like? This is especially important since the spec itself is not clear on the behavior ([radicle-dev/registry-spec#18](https://github.com/radicle-dev/registry-spec/issues/18)).

Does the updated description help? I found the title and the snapshot of the spec to give enough info, let me know how you find it now.

**Comment by geigerzaehler** [comment]
Is it on purpose that we’re not changing the org member list to user IDs instead of accounts? Doesn’t it make sense to make these changes in one go?

**Comment by NunoAlexandre** [comment]
> Is it on purpose that we’re not changing the org member list to user IDs instead of accounts? Doesn’t it make sense to make these changes in one go?

It does make sense yeah, I forgot to add that bit. It's now in.

**Comment by NunoAlexandre** [comment]
Now that I have my hands on this I see further implications to this change. Every time we need to authorize a transaction involving an org (check if x is a member of org), we first need to check whether the said author has an associated user, as that is required to check if they are a member of said org. Meaning, 1) there's a new lookup and therefore new error case, and 2) most of the tests need to updated to register a user for the author used to sign transactions.

**Comment by geigerzaehler** [comment]
Can we create a follow-up ticket for this and #396 to merge into master? In the review of that merge we also need to reconsider the execution strategy.

**Comment by NunoAlexandre** [comment]
@geigerzaehler I have raised the [Upstream features PR](#433) where those critical notes are now placed. Having the PR makes it not only helps to quickly check what's new on that branch but it also explicitly shows what needs to be done before merging, which I find more helpful in this case.
